packages/client/src/runtime/getPrismaClient.ts

```diff
@@ -449,7 +458,9 @@ export function getPrismaClient(config: GetPrismaClientConfig) {
       return 'PrismaClient'
     }
     private getEngine(): Engine {
-      if (this._clientEngineType === ClientEngineType.Library) {
+      if (this._dataProxy === true) {
```

Comment:
`dataProxy = true` is an override of the other engines, as explicitly asked via `--data-proxy`